src/main/java/teammates/sqllogic/core/UsersLogic.java

```diff
@@ -628,6 +715,141 @@ public void resetInstructorGoogleId(String email, String courseId, String google
+                .thenComparing(student -> student.getName()));
+
+        List<String> invalidSectionList = new ArrayList<>();
+        int studentsCount = 1;
+        for (int i = 1; i < mergedList.size(); i++) {
+            Student currentStudent = mergedList.get(i);
+            Student previousStudent = mergedList.get(i - 1);
+            if (currentStudent.getSection().getName().equals(previousStudent.getSection().getName())) {
+                studentsCount++;
+            } else {
+                if (studentsCount > Const.SECTION_SIZE_LIMIT) {
+                    invalidSectionList.add(previousStudent.getSection().getName());
+                }
+                studentsCount = 1;
+            }
+
+            if (i == mergedList.size() - 1 && studentsCount > Const.SECTION_SIZE_LIMIT) {
+                invalidSectionList.add(currentStudent.getSection().getName());
+            }
+        }
+
+        StringJoiner errorMessage = new StringJoiner(" ");
+        for (String section : invalidSectionList) {
+            errorMessage.add(String.format(
+                    ERROR_ENROLL_EXCEED_SECTION_LIMIT,
+                    Const.SECTION_SIZE_LIMIT, section));
+        }
+
+        if (!invalidSectionList.isEmpty()) {
+            errorMessage.add(String.format(
+                    ERROR_ENROLL_EXCEED_SECTION_LIMIT_INSTRUCTION,
+                    Const.SECTION_SIZE_LIMIT));
+        }
+
+        return errorMessage.toString();
+    }
+
+    private String getTeamInvalidityInfo(List<Student> mergedList) {
+        StringJoiner errorMessage = new StringJoiner(" ");
+        mergedList.sort(Comparator.comparing((Student student) -> student.getTeam().getName())
+                .thenComparing(student -> student.getName()));
+
+        List<String> invalidTeamList = new ArrayList<>();
+        for (int i = 1; i < mergedList.size(); i++) {
+            Student currentStudent = mergedList.get(i);
+            Student previousStudent = mergedList.get(i - 1);
+            if (currentStudent.getTeam().getName().equals(previousStudent.getTeam().getName())
+                    && !currentStudent.getSection().getName().equals(previousStudent.getSection().getName())
+                    && !invalidTeamList.contains(currentStudent.getTeam().getName())) {
+
+                errorMessage.add(String.format(ERROR_INVALID_TEAM_NAME,
+                        currentStudent.getTeam().getName(),
+                        previousStudent.getSection().getName(),
+                        currentStudent.getSection().getName()));
+
+                invalidTeamList.add(currentStudent.getTeam().getName());
+            }
+        }
+
+        if (!invalidTeamList.isEmpty()) {
+            errorMessage.add(ERROR_INVALID_TEAM_NAME_INSTRUCTION);
+        }
+
+        return errorMessage.toString();
+    }
+
+    private List<StudentAttributes> getStudentAttributeList(List<Student> students) {
```

Comment:
Is this method needed? Seems that it's only being called in `.getSectionForTeam` above, but that method isn't being called anywhere (it's only being used for datastore)

Reply:
Thanks for catching, will remove both of these unnecessary methods